daemon/cluster/convert/service.go

```diff
@@ -44,6 +44,21 @@ func ServiceFromGRPC(s swarmapi.Service) (types.Service, error) {
 	service.CreatedAt, _ = gogotypes.TimestampFromProto(s.Meta.CreatedAt)
 	service.UpdatedAt, _ = gogotypes.TimestampFromProto(s.Meta.UpdatedAt)
 
+	// If the service is a job, include JobStatus.
+	if curSpec.Mode.ReplicatedJob != nil || curSpec.Mode.GlobalJob != nil {
```

Comment:
Curious, should we just convert everything over as is and let swarmkit validate these kinds of things?

i.e. if `s.Jobstatus != nil` copy it in all cases regardless if other job modes are set?

Reply:
Sure. I can make that change. It shouldn't make a difference one way or another.